examples/complete/main.tf

```diff
@@ -1,109 +1,45 @@
-provider "aws" {
-  region = "${var.region}"
-}
-
-# This `label` is needed to prevent `count can't be computed` errors
-module "label" {
-  source     = "git::https://github.com/cloudposse/terraform-terraform-label.git?ref=master"
-  namespace  = "${var.namespace}"
-  name       = "${var.name}"
-  stage      = "${var.stage}"
-  delimiter  = "${var.delimiter}"
-  attributes = "${var.attributes}"
-  tags       = "${var.tags}"
-  enabled    = "${var.enabled}"
-}
-
-# This `label` is needed to prevent `count can't be computed` errors
-module "cluster_label" {
-  source     = "git::https://github.com/cloudposse/terraform-terraform-label.git?ref=master"
-  namespace  = "${var.namespace}"
-  stage      = "${var.stage}"
-  name       = "${var.name}"
-  delimiter  = "${var.delimiter}"
-  attributes = ["${compact(concat(var.attributes, list("cluster")))}"]
-  tags       = "${var.tags}"
-  enabled    = "${var.enabled}"
-}
-
 locals {
-  # The usage of the specific kubernetes.io/cluster/* resource tags below are required
-  # for EKS and Kubernetes to discover and manage networking resources
-  # https://www.terraform.io/docs/providers/aws/guides/eks-getting-started.html#base-vpc-networking
-  tags = "${merge(var.tags, map("kubernetes.io/cluster/${module.label.id}", "shared"))}"
-}
-
-module "vpc" {
-  source     = "git::https://github.com/cloudposse/terraform-aws-vpc.git?ref=master"
-  namespace  = "${var.namespace}"
-  stage      = "${var.stage}"
-  name       = "${var.name}"
-  attributes = "${var.attributes}"
-  tags       = "${local.tags}"
-  cidr_block = "${var.vpc_cidr_block}"
-}
-
-module "subnets" {
-  source              = "git::https://github.com/cloudposse/terraform-aws-dynamic-subnets.git?ref=master"
-  availability_zones  = ["${var.availability_zones}"]
-  namespace           = "${var.namespace}"
-  stage               = "${var.stage}"
-  name                = "${var.name}"
-  attributes          = "${var.attributes}"
-  tags                = "${local.tags}"
-  region              = "${var.region}"
-  vpc_id              = "${module.vpc.vpc_id}"
-  igw_id              = "${module.vpc.igw_id}"
-  cidr_block          = "${module.vpc.vpc_cidr_block}"
-  nat_gateway_enabled = "true"
+  cluster_name = "${var.namespace}-${var.stage}-${var.name}-cluster"
```

Comment:
Set two locals:

1) for the cluster name. I think the suffix of `-cluster` can be removed. That was 'legacy' when I was testing things out.
2) Print out the set_context command so that a user using this TF template can immediately get using their new cluster!

Reply:
Would be nice to have a variable 'var.cluster_name' .. if that is NULL then generate it this way.

The reason I don't use the `module.cluster.cluster_id` is because that is not known at the plan stage, and some tags are generated from it @aknysh

Reply:
Ok turns out the `-cluster` suffix is created by the cluster module. Couldn't find an exposed parameter for this. This therefore means the local should also have this suffix so everywhere else also gets the same cluster_name!